### ISSUE TITLE
denoising_start does not work normally in pipeline_stable_diffusion_xl_inpaint.py

### ISSUE THREAD

**Issue by CHR-ray** [huggingface/diffusers#4418]
### Describe the bug

I am using StableDiffusionXLInpaintPipeline with stable-diffusion-xl-base-0.9 weight.
And all I use is 
```python
prompt = "2 dogs on the beach"
image = base(
            prompt=prompt,
            num_inference_steps=n_steps,
            image = image,
            denoising_start = denoising_start,
            strength = strength,
            mask_image = mask_img,
        ).images[0]
```
The original image I use is 
![image](https://github.com/huggingface/diffusers/assets/75762846/3fb5e26d-4290-4c82-b605-61f611622d8b)
mask (before resize )is 
![image](https://github.com/huggingface/diffusers/assets/75762846/b62a20d1-418d-4610-a9d1-37312f601395)

when I vary denoising_start, I get 
![image](https://github.com/huggingface/diffusers/assets/75762846/9c2f2e69-d9a3-42ef-afdc-9d00b4b670c1)
when I vary strength, I get
![image](https://github.com/huggingface/diffusers/assets/75762846/a6c1698c-e811-4d0a-80ef-fcc08e410c5c)

You can at least see that different denoising_starts perform almost the same, but different strengths perform differently.

Some **personally** guessing:

When I am reading the code, I notice that the only difference come from 
```
add_noise = True if denoising_start is None else False
latents_outputs = self.prepare_latents(
            batch_size * num_images_per_prompt,
            num_channels_latents,
            height,
            width,
            prompt_embeds.dtype,
            device,
            generator,
            latents,
            image=init_image,
            timestep=latent_timestep,
            is_strength_max=is_strength_max,
            add_noise=add_noise,
            return_noise=True,
            return_image_latents=return_image_latents,
        )
```
and the part that calculate the step
```
timesteps, num_inference_steps = self.get_timesteps(
            num_inference_steps, strength, device, denoising_start=denoising_start if denoising_value_valid else None
        )
```
And in self.prepare_latents, I see
```python
      if latents is None and add_noise:
          noise = randn_tensor(shape, generator=generator, device=device, dtype=dtype)
          # if strength is 1. then initialise the latents to noise, else initial to image + noise
          latents = noise if is_strength_max else self.scheduler.add_noise(image_latents, noise, timestep)
          # if pure noise then scale the initial latents by the  Scheduler's init sigma
          latents = latents * self.scheduler.init_noise_sigma if is_strength_max else latents
      elif add_noise:
          noise = latents.to(device)
          latents = noise * self.scheduler.init_noise_sigma
      else:
          noise = randn_tensor(shape, generator=generator, device=device, dtype=dtype)
          latents = image_latents.to(device)
```
I notice that in the first and second if branch, we all did ``* self.scheduler.init_noise_sigma`` to the latents, but when 
add_noise is False (denoising_start is not None), we did not times this constant. I am not sure, but is this part ``self.scheduler.init_noise_sigma `` cause this?

### Reproduction

```python
prompt = "2 dogs on the beach"
image = base(
            prompt=prompt,
            num_inference_steps=n_steps,
            image = image,
            denoising_start = denoising_start,
            strength = strength,
            mask_image = mask_img,
        ).images[0]
```
with image and mask given above

### System Info

Package                      Version           Editable project location
---------------------------- ----------------- ------------------------------------
absl-py                      1.4.0
accelerate                   0.20.3
addict                       2.4.0
aiofiles                     23.1.0
aiohttp                      3.8.4
aiosignal                    1.3.1
albumentations               1.3.0
altair                       4.2.2
anyio                        3.6.2
argon2-cffi                  21.3.0
argon2-cffi-bindings         21.2.0
arrow                        1.2.3
asttokens                    2.2.1
astunparse                   1.6.3
async-lru                    2.0.3
async-timeout                4.0.2
attrs                        22.2.0
Babel                        2.12.1
backcall                     0.2.0
beautifulsoup4               4.12.2
bitsandbytes                 0.35.0
bleach                       6.0.0
boto3                        1.28.3
botocore                     1.31.3
cachetools                   5.3.0
certifi                      2023.5.7
cffi                         1.15.1
charset-normalizer           3.0.1
chumpy                       0.70
click                        8.1.3
cmake                        3.25.2
colorama                     0.4.6
coloredlogs                  15.0.1
comm                         0.1.3
contourpy                    1.0.7
cycler                       0.11.0
Cython                       0.29.36
debugpy                      1.6.7
decorator                    5.1.1
defusedxml                   0.7.1
diffusers                    0.19.0.dev0       
easydict                     1.10
easygui                      0.98.3
einops                       0.6.0
entrypoints                  0.4
exceptiongroup               1.1.2
executing                    1.2.0
fairscale                    0.4.13
fastapi                      0.92.0
fastjsonschema               2.17.1
ffmpy                        0.3.0
filelock                     3.9.0
filterpy                     1.4.5
flatbuffers                  23.1.21
fonttools                    4.38.0
fqdn                         1.5.1
freetype-py                  2.4.0
frozenlist                   1.3.3
fsspec                       2023.1.0
ftfy                         6.1.1
gast                         0.4.0
google-auth                  2.16.1
google-auth-oauthlib         0.4.6
google-pasta                 0.2.0
gradio                       3.16.2
grpcio                       1.51.3
h11                          0.14.0
h5py                         3.8.0
httpcore                     0.16.3
httpx                        0.23.3
huggingface-hub              0.16.4
humanfriendly                10.0
idna                         3.4
imageio                      2.25.1
importlib-metadata           6.0.0
invisible-watermark          0.2.0
ipykernel                    6.24.0
ipython                      8.14.0
isoduration                  20.11.0
jedi                         0.18.2
Jinja2                       3.1.2
jmespath                     1.0.1
joblib                       1.2.0
json-tricks                  3.17.1
json5                        0.9.14
jsonpointer                  2.4
jsonschema                   4.17.3
jsonschema-specifications    2023.6.1
jupyter_client               8.3.0
jupyter_core                 5.3.1
jupyter-events               0.6.3
jupyter-lsp                  2.2.0
jupyter_server               2.7.0
jupyter_server_terminals     0.4.4
jupyterlab                   4.0.3
jupyterlab-pygments          0.2.2
jupyterlab_server            2.23.0
keras                        2.10.0
Keras-Preprocessing          1.1.2
kiwisolver                   1.4.4
lazy_loader                  0.3
libclang                     15.0.6.1
lightning-utilities          0.7.1
linkify-it-py                2.0.0
lion-pytorch                 0.0.7
llvmlite                     0.40.1
Markdown                     3.4.1
markdown-it-py               2.2.0
MarkupSafe                   2.1.2
matplotlib                   3.7.0
matplotlib-inline            0.1.6
mdit-py-plugins              0.3.4
mdurl                        0.1.2
mistune                      3.0.1
mmcv-full                    1.7.1
mmpose                       0.28.0
model-index                  0.1.11
mpmath                       1.3.0
multidict                    6.0.4
munkres                      1.1.4
mypy-extensions              1.0.0
nbclient                     0.8.0
nbconvert                    7.6.0
nbformat                     5.9.1
nest-asyncio                 1.5.6
networkx                     3.0
notebook_shim                0.2.3
numba                        0.57.1
numpy                        1.24.2
oauthlib                     3.2.2
onnx                         1.14.0
onnxruntime                  1.15.1
opencv-python                4.7.0.68
opencv-python-headless       4.7.0.72
opendatalab                  0.0.9
openmim                      0.3.9
opt-einsum                   3.3.0
ordered-set                  4.1.0
orjson                       3.8.6
overrides                    7.3.1
packaging                    23.0
pandas                       1.5.3
pandocfilters                1.5.0
parso                        0.8.3
pexpect                      4.8.0
pickleshare                  0.7.5
Pillow                       9.4.0
pip                          23.2
platformdirs                 3.9.1
plyfile                      1.0
prometheus-client            0.17.1
prompt-toolkit               3.0.39
protobuf                     3.19.6
psutil                       5.9.4
ptyprocess                   0.7.0
pure-eval                    0.2.2
pyasn1                       0.4.8
pyasn1-modules               0.2.8
pycocotools                  2.0.6
pycparser                    2.21
pycryptodome                 3.17
pydantic                     1.10.5
pydub                        0.25.1
pyglet                       1.5.27
Pygments                     2.15.1
pynvml                       11.5.0
PyOpenGL                     3.1.0
pyparsing                    3.0.9
pyre-extensions              0.0.23
pyrender                     0.1.45
pyrsistent                   0.19.3
python-dateutil              2.8.2
python-json-logger           2.0.7
python-multipart             0.0.5
pytorch-lightning            1.9.0
pytz                         2022.7.1
PyWavelets                   1.4.1
PyYAML                       6.0
pyzmq                        25.1.0
qudida                       0.0.4
referencing                  0.29.1
regex                        2022.10.31
requests                     2.28.2
requests-oauthlib            1.3.1
rfc3339-validator            0.1.4
rfc3986                      1.5.0
rfc3986-validator            0.1.1
rich                         13.4.2
rpds-py                      0.8.10
rsa                          4.9
s3transfer                   0.6.1
safetensors                  0.3.1
scikit-image                 0.19.3
scikit-learn                 1.2.1
scipy                        1.10.1
seaborn                      0.12.2
Send2Trash                   1.8.2
setuptools                   67.8.0
six                          1.16.0
smplx                        0.1.28
sniffio                      1.3.0
soupsieve                    2.4.1
stack-data                   0.6.2
starlette                    0.25.0
supervision                  0.11.1
sympy                        1.12
tabulate                     0.9.0
tensorboard                  2.10.1
tensorboard-data-server      0.6.1
tensorboard-plugin-wit       1.8.1
tensorboardX                 2.6.1
tensorflow                   2.10.1
tensorflow-estimator         2.10.0
tensorflow-io-gcs-filesystem 0.31.0
termcolor                    2.2.0
terminado                    0.17.1
threadpoolctl                3.1.0
tifffile                     2023.2.3
timm                         0.6.12
tinycss2                     1.2.1
tokenizers                   0.13.2
tomli                        2.0.1
toolz                        0.12.0
torch                        1.12.1+cu116
torchaudio                   0.11.0+rocm4.5.2
torchgeometry                0.1.2
torchmetrics                 0.11.1
torchvision                  0.13.1+cu116
tornado                      6.3.2
tqdm                         4.64.1
traitlets                    5.9.0
transformers                 4.31.0
trimesh                      3.22.4
triton                       2.0.0.dev20221202
typing_extensions            4.5.0
typing-inspect               0.8.0
tzdata                       2023.3
uc-micro-py                  1.0.1
uri-template                 1.3.0
urllib3                      1.26.14
uvicorn                      0.20.0
wcwidth                      0.2.6
webcolors                    1.13
webencodings                 0.5.1
websocket-client             1.6.1
websockets                   10.4
Werkzeug                     2.2.3
wheel                        0.38.4
wrapt                        1.14.1
xtcocotools                  1.13
yacs                         0.1.8
yapf                         0.40.1
yarl                         1.8.2
zipp                         3.15.0

**Comment by AmericanPresidentJimmyCarter** [comment]
This is not how you use `denoising_start`.

You need to generate first image from the base model with `denoising_end` declared, then pass that image as a latent to the second refiner model with `denoising_start` declared.

https://huggingface.co/docs/diffusers/api/pipelines/stable_diffusion/stable_diffusion_xl#1-ensemble-of-expert-denoisers

**Comment by sayakpaul** [comment]
Closing with @AmericanPresidentJimmyCarter's suggestions. Feel free to reopen.
